packages/astro/src/core/render-context.ts

```diff
@@ -394,11 +395,37 @@ export class RenderContext {
 				}
 				return renderContext.session;
 			},
-			insertDirective(_payload) {
+			insertDirective(payload) {
+				if (!!pipeline.manifest.csp === false) {
```

Comment:
```suggestion
				if (!pipeline.manifest.csp) {
```
You're coercing to boolean and then comparing to false, so it's equivalent to this, which is simpler